io.openems.edge.ess.generic/src/io/openems/edge/ess/generic/symmetric/GenericManagedSymmetricEssImpl.java

```diff
@@ -0,0 +1,248 @@
+		immediate = true, //
+		configurationPolicy = ConfigurationPolicy.REQUIRE, //
+		property = { //
+				EventConstants.EVENT_TOPIC + "=" + EdgeEventConstants.TOPIC_CYCLE_AFTER_PROCESS_IMAGE //
+		} //
+)
+public class GenericManagedSymmetricEssImpl extends AbstractOpenemsComponent implements GenericManagedSymmetricEss,
+		ManagedSymmetricEss, SymmetricEss, OpenemsComponent, EventHandler, StartStoppable {
+
+	@Reference
+	private Power power;
+
+	@Reference
+	private ConfigurationAdmin cm;
+
+	@Reference(policy = ReferencePolicy.STATIC, policyOption = ReferencePolicyOption.GREEDY, cardinality = ReferenceCardinality.MANDATORY)
+	private ManagedSymmetricBatteryInverter batteryInverter;
+
+	@Reference(policy = ReferencePolicy.STATIC, policyOption = ReferencePolicyOption.GREEDY, cardinality = ReferenceCardinality.MANDATORY)
+	private Battery battery;
+
+	private final Logger log = LoggerFactory.getLogger(GenericManagedSymmetricEssImpl.class);
+
+	/**
+	 * Manages the {@link State}s of the StateMachine.
+	 */
+	private final StateMachine<State, Context> stateMachine = new StateMachine<>(State.UNDEFINED);
+
+	/**
+	 * Helper wrapping class to handle everything related to Channels.
+	 */
+	private final ChannelHandler channelHandler = new ChannelHandler(this);
+
+	private Config config = null;
+
+	public GenericManagedSymmetricEssImpl() {
+		super(//
+				OpenemsComponent.ChannelId.values(), //
+				StartStoppable.ChannelId.values(), //
+				SymmetricEss.ChannelId.values(), //
+				ManagedSymmetricEss.ChannelId.values(), //
+				GenericManagedSymmetricEss.ChannelId.values() //
+		);
+	}
+
+	@Activate
+	void activate(ComponentContext context, Config config) {
+		super.activate(context, config.id(), config.alias(), config.enabled());
+
+		// update filter for 'BatteryInverter'
+		if (OpenemsComponent.updateReferenceFilter(this.cm, this.servicePid(), "batteryInverter",
+				config.batteryInverter_id())) {
+			return;
+		}
+
+		// update filter for 'Battery'
+		if (OpenemsComponent.updateReferenceFilter(this.cm, this.servicePid(), "battery", config.battery_id())) {
+			return;
+		}
+
+		this.channelHandler.activate(this.battery, this.batteryInverter);
+		this.config = config;
+	}
+
+	@Deactivate
+	protected void deactivate() {
+		this.channelHandler.deactivate();
+		super.deactivate();
+	}
+
+	@Override
+	public void handleEvent(Event event) {
+		if (!this.isEnabled()) {
+			return;
+		}
+		switch (event.getTopic()) {
+
+		case EdgeEventConstants.TOPIC_CYCLE_AFTER_PROCESS_IMAGE:
+			this.handleStateMachine();
+			break;
+		}
+	}
+
+	/**
+	 * Handles the State-Machine.
+	 */
+	private void handleStateMachine() {
+		// Store the current State
+		this.channel(GenericManagedSymmetricEss.ChannelId.STATE_MACHINE)
+				.setNextValue(this.stateMachine.getCurrentState());
+
+		// Initialize 'Start-Stop' Channel
+		this._setStartStop(StartStop.UNDEFINED);
+
+		// Prepare Context
+		Context context = new Context(this, this.battery, this.batteryInverter, this.config);
```

Comment:
No, this was on purpose. Anyway the Context is rebuilt on every Cycle so in case the references get dropped, they can be cleaned by garbage collection in the upcoming Cycle.